stix2validator/validator.py

```diff
@@ -110,8 +111,38 @@ def as_json(self):
         return json.dumps(self.as_dict())
 
 
-class ValidationResults(BaseResults):
-    """Results of JSON schema validation.
+class FileValidationResults(BaseResults):
+    """
+    Represents validation results for a file.  This entails potentially
+    several STIX object results, since a file may contain a list of STIX
+    objects.
+    """
+    def __init__(self, is_valid=False, filepath=None, object_results=None, fatal=None):
+        """
+        Initialize this instance.
+        :param is_valid: Whether the overall result is valid
+        :param filepath: Which file was validated
+        :param object_results: Individual object validation results
+        :param fatal: A non-validation-related fatal error
+        """
+        super(FileValidationResults, self).__init__(is_valid)
+        self.filepath = filepath
+        self.object_results = object_results or []
+        self.fatal = fatal
+
+    def as_dict(self):
+        d = super(FileValidationResults, self).as_dict()
+        d.update(
+            filepath=self.filepath,
+            object_results=[object_result.as_dict() for object_result in self.object_results],
+            fatal=self.fatal.as_dict()
+        )
```

Comment:
Are we just making this results dict structure up?  What is it used for?

Reply:
I think it's a holdover inherited from [STIXProject/stix-validator](https://github.com/STIXProject/stix-validator). However, it's used in `*Results.as_json()` which may be helpful for some users.

Reply:
Are there tests for the JSON format? It would be good to have a separate page in the documentation (once we create it) for the format. I can envision it being used in web applications that do STIX validation on the back-end.